src/zarr/abc/store.py

```diff
@@ -145,3 +147,28 @@ def list_dir(self, prefix: str) -> AsyncGenerator[str, None]:
         AsyncGenerator[str, None]
         """
         ...
+
+
+@runtime_checkable
+class ByteGetter(Protocol):
+    async def get(
+        self, byte_range: Optional[Tuple[int, Optional[int]]] = None
+    ) -> Optional[BytesLike]: ...
+
+
+@runtime_checkable
+class ByteSetter(Protocol):
+    async def get(
+        self, byte_range: Optional[Tuple[int, Optional[int]]] = None
+    ) -> Optional[BytesLike]: ...
+
+    async def set(self, value: BytesLike, byte_range: Optional[Tuple[int, int]] = None) -> None: ...
+
+    async def delete(self) -> None: ...
+
+
+async def set_or_delete(byte_setter: ByteSetter, value: BytesLike | None) -> None:
```

Comment:
do we still need this if we consider writing `{}` to `.zattrs` functionally equivalent to deleting `.zattrs`?

Reply:
This is still required for chunks.